### ISSUE TITLE
Authentication-Results values not shown

### ISSUE THREAD

**Issue by alevesely** [lieser/dkim_verifier#466]
I have set both _Verify DKIM signatures_ and _Read Authentication-Results header_.  However, a message having:
``` email
Authentication-Results: wmail.tana.it;
  spf=pass smtp.mailfrom=mailop.org;
  dkim=pass reason="Original-From: transformed" header.d=dcrocker.net;
  dmarc=pass header.from=mailop.org;
  arc=fail (1 set(s)) smtp.remote-ip=91.132.147.157
```
displays `DKIM   Invalid (E-Mail was modified)` despite the signature was previously authenticated.  My understanding was that  _Read Authentication-Results header_ meant to skip performing a verification when a trusted A-R already has a result.  Or is that A-R invalid for some reason?  I don't see references to that in the error console, which starts like so:

```console
DKIM_Verifier.AuthVerifier	ERROR	 Ignoring error in parsing of ARH DKIM_Error: Parsing error
    match moz-extension://694bf5d4-0ccd-4013-88d6-a4fdb3a218aa/modules/arhParser.mjs.js:354
    parseResInfo moz-extension://694bf5d4-0ccd-4013-88d6-a4fdb3a218aa/modules/arhParser.mjs.js:188
    parse moz-extension://694bf5d4-0ccd-4013-88d6-a4fdb3a218aa/modules/arhParser.mjs.js:158
    getARHResult moz-extension://694bf5d4-0ccd-4013-88d6-a4fdb3a218aa/modules/authVerifier.mjs.js:265
    verify moz-extension://694bf5d4-0ccd-4013-88d6-a4fdb3a218aa/modules/authVerifier.mjs.js:177
[authVerifier.mjs.js:267:8](moz-extension://694bf5d4-0ccd-4013-88d6-a4fdb3a218aa/modules/authVerifier.mjs.js)
DKIM_Verifier.AuthVerifier	DEBUG	 ARH result: 
Object { version: "3.1", dkim: [], spf: [], dmarc: [], bimiIndicator: undefined }
[authVerifier.mjs.js:348:6](moz-extension://694bf5d4-0ccd-4013-88d6-a4fdb3a218aa/modules/authVerifier.mjs.js)
DKIM_Verifier.Verifier	DEBUG	 1 DKIM-Signatures found. [verifier.mjs.js:1491:8](moz-extension://694bf5d4-0ccd-4013-88d6-a4fdb3a218aa/modules/dkim/verifier.mjs.js)
DKIM_Verifier.Verifier	DEBUG	 Verifying DKIM-Signature 1 ... [verifier.mjs.js:1506:9](moz-extension://694bf5d4-0ccd-4013-88d6-a4fdb3a218aa/modules/dkim/verifier.mjs.js)
```
The signature is indeed invalid, according to the spec.  The A-R value is based on [a method to undo the transformation done by a mailing list](https://datatracker.ietf.org/doc/html/draft-vesely-dmarc-mlm-transform) which could verify the author's signature.

Why isn't the A-R value accepted by DKIM Verifier?

**Comment by ell1e** [comment]
Did you try enabling parsing the `Authentication-Results` header in the addon preferences? I believe it's disabled by default, which could explain what you're seeing.

**Comment by lieser** [comment]
> I don't see references to that in the error console, which starts like so:

Yeah the log entry could be more clear, but the `Ignoring error in parsing of ARH` basically means the ARH the parsing fails for is ignored.

> Why isn't the A-R value accepted by DKIM Verifier?

The problem is the `(1 set(s))` comment, more specific the `(s)` parts in it. Will investigate if that is allowed by the RFC.

**Comment by alevesely** [comment]
I thought being a comment you could just skip it entirely.  My reckoning is that comments are allowed more or less everywhere.  In this case ([RFC8601](https://datatracker.ietf.org/doc/html/rfc8601#section-2.2):
```abnf
     resinfo = [CFWS] ";" methodspec [ CFWS reasonspec ]
               [ CFWS 1*propspec ]
```
The `(1 set(s))` could be the CFWS before the propspec.  Recursivity derives from CFWS definition ([RFC5322](https://datatracker.ietf.org/doc/html/rfc5322#section-3.2.2):
```abnf
   ccontent        =   ctext / quoted-pair / comment
   comment         =   "(" *([FWS] ccontent) [FWS] ")"
   CFWS            =   (1*([FWS] comment) [FWS]) / FWS
```

**Comment by lieser** [comment]
This was indeed a bug/limitation of the add-on, and strictly speaking still is.

The problem is that the `comment` definition is recursive, i.e. `comment` is allowed to appear inside a `comment` via the `ccontent`. Currently regular expressions are used for most of the parsing, which do not support such recursion.
I now manually unrolled it to support up to 3 levels of comments, which is hopefully enough.

**Comment by alevesely** [comment]
>  I now manually unrolled it to support up to 3 levels of comments, which is hopefully enough.

Thank you.  I'm looking forward to getting a new release.

**Comment by lieser** [comment]
I plan to release 5,5,0 in 1-2 weeks, depending on how fast the translations will be.

You could also already pack it yourself, see https://github.com/lieser/dkim_verifier?tab=readme-ov-file#packing-the-add-on.

**Comment by alevesely** [comment]
Thanks, I'll wait for the translations.
